server/server.js

```diff
@@ -142,24 +146,49 @@ Server.prototype._handleClientMessage = function(client, data) {
     return;
   }
 
-  logging.info('#client.message - received', client.id, message,
-     (this.resources[message.to] ? 'exists' : 'not instantiated'),
-     (this.subs[message.to] ? 'is subscribed' : 'not subscribed')
-    );
+  if (!this._clientDataPersist(client.id, message)) {
+    return;
+  }
+
+  this._resourceMessageHandle(client, message);
+};
+
+// Initialize a client, and persist messages where required
+Server.prototype._clientDataPersist = function (id, message) {
+  // Sync the client name to the current client id
+  if (message.op == 'name_id_sync') {
+    this._clientInit(message);
+    this.clientVersion = message.client_version;
```

Comment:
I had made a note to myself about this, and then forgot to follow up on it - good catch.  

When a client sends the *name_id_sync* message, the client version is fixed for that current session, so I could use an existing hash, keyed on id (or name), or create a new hash.  I'll look into it.